### ISSUE TITLE
[Bug] Asynchronous Actions not working as expected

### ISSUE THREAD

**Issue by Avejack** [kamranahmedse/driver.js#121]
Hey,

In your [Asynchronous Actions - Demo](https://github.com/kamranahmedse/driver.js#asynchronous-actions--demo) you have this comment:

> // Perform some action or create the element to move to
      // And then move to that element

but i tried it and it didn't work. I took a look in your code and here is the problem:
When defining the steps via "defineSteps" you iterate through the array and ignore all step definitions where your function "prepareElementFromStep" does not return an element. Therefore in your "moveNext" function "nextStep" may not be defined or at least is not present in the stack.

What i would suggest:
add steps to steps-array regardless if the element exists (this would also help with the floating dialog feature request: #67 ) or not and in every "moveNext"-function call test for the element. If there is none: Display a floating Dialog.

Sorry if i missed something!

**Comment by remibd** [comment]
Hi !

Having encountered the same problem today, I second that!

When `defineSteps()` is called, it stops if it encounters a step with an element does not exist yet.
It is problematic when one wants to highlight an element that did not exist yet at step definition, and will probably crash if the element to be highlighted has been removed from the DOM after step definition.

The solution could be to delay [the call to `document.querySelector()`](https://github.com/kamranahmedse/driver.js/blob/6dc270de19d75b314d3b2c5ca6368c66fefd62de/src/index.js#L365) to [the actual `moveNext()` function](https://github.com/kamranahmedse/driver.js/blob/6dc270de19d75b314d3b2c5ca6368c66fefd62de/src/index.js#L254), but I’m not sure I understand the code enough to do such a modification myself.

A **workaround** is to reload the steps in a `onNext()` handler *after* the offending nodes have been created, like that:
```javascript
onNext: () => {
    if (!driver.hasNextStep()) {
        driver.preventMove();
        driver.defineSteps(steps);
        driver.moveNext();
    }
}
```

Hope it helps :slightly_smiling_face:

**Comment by oliverbienert** [comment]
Hello, first thank you for the good work you've done!
However, I stumbled over the same problem. If you check at the beginning of a tour that all elements within exist, the above comment (Taken from the official documentation) is highly misleading, because it suggests that it is possible to create an element on the fly. But that very case is not possible when you expect all elements to be existing at the beginning.
Is there any news on this?

**Comment by jawn-ha** [comment]
> Hello, first thank you for the good work you've done!
> However, I stumbled over the same problem. If you check at the beginning of a tour that all elements within exist, the above comment (Taken from the official documentation) is highly misleading, because it suggests that it is possible to create an element on the fly. But that very case is not possible when you expect all elements to be existing at the beginning.
> Is there any news on this?

Hi, Did you solve it? I have same problem

**Comment by kamranahmedse** [comment]
We just released [v1.0](https://github.com/kamranahmedse/driver.js/releases/tag/1.0.3) of driver.js which has been rewritten from the ground up. This issue should no longer exist. Please have a [look at the docs](https://driverjs.com/) and do let me know if you face any issues.
